ypathgen/pathgen.go

```diff
@@ -740,6 +766,74 @@ func generateChildConstructorsForLeafOrContainer(methodBuf *strings.Builder, fie
 	return errors
 }
 
+// generateChildConstructorsForListBuilderFormat writes into methodBuf the
+// child constructor method snippets for the list represented by listAttr using
+// the builder API format. fieldData contains the childConstructor template
+// output information for if the node were a container (which contains a subset
+// of the basic information required for the list constructor methods).
+func generateChildConstructorsForListBuilderFormat(methodBuf *strings.Builder, listAttr *ygen.YangListAttr, fieldData goPathFieldData, isUnderFakeRoot bool, schemaStructPkgAlias string) []error {
+	var errors []error
+	// List of function parameters as would appear in the method definition.
+	keyParams, err := makeKeyParams(listAttr, schemaStructPkgAlias)
+	if err != nil {
+		return append(errors, err)
+	}
+	keyN := len(keyParams)
+
+	// Initialize ygot.NodePath's key list with wildcard values.
+	var keyEntryStrs []string
+	for i := 0; i != keyN; i++ {
+		keyEntryStrs = append(keyEntryStrs, fmt.Sprintf(`"%s": "*"`, keyParams[i].name))
+	}
+	fieldData.KeyEntriesStr = strings.Join(keyEntryStrs, ", ")
+
+	// There are no initial key parameters for the builder API.
+	fieldData.KeyParamListStr = ""
+
+	// Set the child type to be the wildcard version.
+	fieldData.TypeName += WildcardSuffix
+
+	// Add Builder suffix to the child constructor method name.
+	fieldData.MethodName += BuilderCtorSuffix
+
+	// Generate builder constructor method for non-wildcard version of parent struct.
+	if err := goPathChildConstructorTemplate.Execute(methodBuf, fieldData); err != nil {
+		errors = append(errors, err)
+	}
+
+	// The root node doesn't have a wildcard version of itself.
+	if !isUnderFakeRoot {
+		// Generate builder constructor method for wildcard version of parent struct.
+		fieldData.Struct.TypeName += WildcardSuffix
+		if err := goPathChildConstructorTemplate.Execute(methodBuf, fieldData); err != nil {
+			return append(errors, err)
+		}
+	}
+
+	// Generate key-builder methods for the wildcard version of the PathStruct.
+	// Although non-wildcard PathStruct is unnecessary, it is kept for generation simplicity.
+	for i := 0; i != keyN; i++ {
+		if err := goKeyBuilderTemplate.Execute(methodBuf,
+			struct {
+				MethodName    string
+				TypeName      string
+				KeySchemaName string
+				KeyParamType  string
+				KeyParamName  string
+			}{
+				MethodName:    BuilderKeyPrefix + keyParams[i].varName,
+				TypeName:      fieldData.TypeName,
+				KeySchemaName: keyParams[i].name,
+				KeyParamName:  keyParams[i].varName,
+				KeyParamType:  keyParams[i].typeName,
+			}); err != nil {
+			return append(errors, err)
```

Comment:
deliberately returning immediately rather than accumulate the errors for all the keys?

Reply:
Thanks, nice catch!